src/Tools/dotnet-user-jwts/src/Helpers/DevJwtCliHelpers.cs

```diff
@@ -0,0 +1,204 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.IdentityModel.Tokens.Jwt;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Text.Json;
+using System.Xml.Linq;
+using System.Xml.XPath;
+using Microsoft.Extensions.CommandLineUtils;
+using Microsoft.Extensions.Configuration;
+
+namespace Microsoft.AspNetCore.Authentication.JwtBearer.Tools;
+
+internal static class DevJwtCliHelpers
+{
+    public static string GetUserSecretsId(string projectFilePath)
+    {
+        var projectDocument = XDocument.Load(projectFilePath, LoadOptions.PreserveWhitespace);
+        var existingUserSecretsId = projectDocument.XPathSelectElements("//UserSecretsId").FirstOrDefault();
+
+        if (existingUserSecretsId == null)
+        {
+            return null;
+        }
+
+        return existingUserSecretsId.Value;
+    }
+
+    public static string GetProject(string projectPath = null)
+    {
+        if (projectPath is not null)
+        {
+            return projectPath;
+        }
+
+        var csprojFiles = Directory.GetFiles(Directory.GetCurrentDirectory(), "*.csproj");
+        if (csprojFiles is [var path])
+        {
+            return path;
+        }
+        return null;
+    }
+
+    public static byte[] GetOrCreateSigningKeyMaterial(string userSecretsId)
+    {
+        var projectConfiguration = new ConfigurationBuilder()
+            .AddUserSecrets(userSecretsId)
+            .Build();
+
+        var signingKeyMaterial = projectConfiguration[DevJwtsDefaults.SigningKeyConfigurationKey];
+
+        var keyMaterial = new byte[DevJwtsDefaults.SigningKeyLength];
+        if (signingKeyMaterial is not null && Convert.TryFromBase64String(signingKeyMaterial, keyMaterial, out var bytesWritten) && bytesWritten == DevJwtsDefaults.SigningKeyLength)
+        {
+            return keyMaterial;
+        }
+
+        return CreateSigningKeyMaterial(userSecretsId);
+    }
+
+    public static byte[] CreateSigningKeyMaterial(string userSecretsId, bool reset = false)
+    {
+        // Create signing material and save to user secrets
+        var newKeyMaterial = System.Security.Cryptography.RandomNumberGenerator.GetBytes(DevJwtsDefaults.SigningKeyLength);
+
+        string secretsFilePath;
+        if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+        {
+            secretsFilePath = Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData), "Microsoft", "UserSecrets", userSecretsId, "secrets.json");
+        }
+        else
+        {
+            secretsFilePath = Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.UserProfile), "microsoft", "usersecrets", userSecretsId, "secret.json");
+        }
```

Comment:
```suggestion
        var secretsFilePath = PathHelper.GetSecretsPathFromSecretsId(userSecretsId);
```

This comes from Microsoft.Extensions.Configuration.UserSecrets, so we'll need to add a using. This is what user-secrets does and has some subtle behaviors like using the `DOTNET_USER_SECRETS_FALLBACK_DIR`.